src/command.rs

```diff
@@ -1,36 +1,124 @@
-use clap::{AppSettings, Args, Parser, ValueEnum};
+// TODO: docstrings for everything!!!
 
-#[derive(Parser)]
-#[clap(name = "cargo")]
-#[clap(bin_name = "cargo")]
+use clap::{Args, Parser, Subcommand};
+
+#[derive(Parser, Debug)]
+#[command(name = "cargo")]
+#[command(bin_name = "cargo")]
 pub enum Cargo {
-    #[clap(name = "3ds")]
+    #[command(name = "3ds")]
     Input(Input),
 }
 
-#[derive(Args)]
-#[clap(about)]
-#[clap(global_setting(AppSettings::AllowLeadingHyphen))]
+#[derive(Args, Debug)]
+#[command(version, about)]
 pub struct Input {
-    #[clap(value_enum)]
+    /// The cargo command to run. This command will be forwarded to the real
+    /// `cargo` with the appropriate arguments for a 3DS executable.
+    #[command(subcommand)]
     pub cmd: CargoCommand,
-    pub cargo_opts: Vec<String>,
+
+    /// Don't actually run any commands, just echo them to the console.
+    /// This is mostly intended for testing.
+    #[arg(long, hide = true)]
+    pub dry_run: bool,
```

Comment:
Will this be removed when the PR is published?

Reply:
Maybe not? I was thinking this might be a good way to test the CLI works as expected, but maybe also it would be better to do as a separate change, or better to test the CLI against a real crate in a CI environment like we have in `ctru-rs`.

Reply:
Yeah, should be a separate PR.